auth/server/base/src/main/java/org/wildfly/security/authz/RegexRoles.java

```diff
@@ -0,0 +1,82 @@
+package org.wildfly.security.authz;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+import java.util.stream.StreamSupport;
+
+import static org.wildfly.security.auth.server._private.ElytronMessages.log;
+
+/**
+ * A regex roles.
+ * <p>
+ * Roles that are checked against regex and if matching are mapped with given replacement pattern. Regex can capture groups that replacement pattern can make use of.
+ *
+ * @author <a href="mailto:dvilkola@redhat.com">Diana Vilkolakova</a>
+ */
+public class RegexRoles implements Roles {
```

Comment:
We might want to make this package private so that it's consistent with the other ```Roles``` implementations.

Reply:
Fixed